### PR TITLE
bug/21: Fixed free port detection

### LINKED ISSUE #21: Free port detection broken
The free port detection fails if the container does not contain `netstat`.

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #21

### CODE REVIEW

File: doc/changes/changes_0.1.1.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# exasol-test-setup-abstraction-java 0.1.1, released 2021-07-13
+
+Code name: Fixed Free Port Detection
+
+## Bug Fixes:
```
Comment (morazow) [inline doc/changes/changes_0.1.1.md:5]:
```suggestion
## Bug Fixes
```

File: src/main/java/com/exasol/exasoltestsetup/SshConnection.java [hunk 2]
```diff
@@ -123,19 +123,22 @@ public SshExecution runCommand(final String command) {
...
+                "with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:\n" + //
+                "    s.bind(('', 0))\n" + //
+                "    s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)\n" + //
+                "    print(s.getsockname()[1])\n\n" + //
+                "HEREDOC").whenFinished().assertExitCodeIsZero().getStdout();
+        try {
+            return Integer.parseInt(portOutput.trim());
+        } catch (final NumberFormatException exception) {
+            throw new IllegalStateException(ExaError.messageBuilder("F-ETAJ-25")
+                    .message("Failed find free port on server. The python script had an invalid output: {{output}}.",
```
Comment (morazow) [inline src/main/java/com/exasol/exasoltestsetup/SshConnection.java:138]:
```suggestion
                    .message("Failed to find free port on server. The python script had an invalid output: {{output}}.",
```

File: src/main/java/com/exasol/exasoltestsetup/SshConnection.java [hunk 3]
```diff
@@ -123,19 +123,22 @@ public SshExecution runCommand(final String command) {
...
-            final int usedPort = Integer.parseInt(parts[parts.length - 1]);
-            maxUsedPort = Math.max(maxUsedPort, usedPort);
+        final String portOutput = runCommand("python3 <<HEREDOC\n" + //
+                "import socket\n" + //
+                "from contextlib import closing\n" + //
+                "with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:\n" + //
+                "    s.bind(('', 0))\n" + //
+                "    s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)\n" + //
+                "    print(s.getsockname()[1])\n\n" + //
+                "HEREDOC").whenFinished().assertExitCodeIsZero().getStdout();
```
Comment (morazow) [inline src/main/java/com/exasol/exasoltestsetup/SshConnection.java:133]: I would suggest to extract python script to a method.

**Review by morazow — APPROVED** [review]
Added some suggestions, but looks good!

File: src/main/java/com/exasol/exasoltestsetup/SshConnection.java (hunk 3 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/exasoltestsetup/SshConnection.java:133]: I extracted a constant

**Review by morazow — APPROVED** [review]

**Pull request merged**